### PR TITLE
Bug 1202592 - Activity chooser leaks settings observer

### PR DESCRIPTION
**Pull request by fcampo**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0da4d8aedb26f6e1a83ab927972350d809412e9a

File: shared/js/default_activity_helper.js [hunk 1]
```diff
@@ -132,7 +132,7 @@
     ];
 
   var DefaultActivityHelper = {
-
+    helper: null,
```
Comment (etiennesegonzac) [inline shared/js/default_activity_helper.js]: nit: you can call it `_helper` to hint that this is internal

File: shared/js/default_activity_helper.js [hunk 2]
```diff
@@ -159,8 +159,10 @@
         if (!config) {
           resolve(null);
         }
-
-        SettingsHelper(config.settingsId, null).get((value) => {
+        if (!this.helper) {
+          this.helper = SettingsHelper(config.settingsId, null);
+        }
```
Comment (etiennesegonzac) [inline shared/js/default_activity_helper.js]: nit: maybe you can extract this in a `getHelper` method to avoid the repetition

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a4ef79223ca98ea0ba05496f21bb2fb1067eef3e

**Pull request merged**